### PR TITLE
ENH: Support datetime.timezone objects

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Introduced in Python 3.2. First pass is to evaluate what tests fail when included in the tz_aware_fixture

### CODE REVIEW

File: pandas/_libs/tslibs/timezones.pyx [hunk 1]
```diff
@@ -27,7 +28,7 @@ cdef int64_t NPY_NAT = get_nat()
 # ----------------------------------------------------------------------
 
 cpdef inline bint is_utc(object tz):
-    return tz is UTC or isinstance(tz, _dateutil_tzutc)
+    return tz is UTC or isinstance(tz, _dateutil_tzutc) or tz is timezone.utc
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timezones.pyx]: Consider “cdef tzinfo utc_stdlib = timezone.utc” at module level.  Much faster lookup for frequently-done check.

Comment (mroeschke) [reply pandas/_libs/tslibs/timezones.pyx]: Wasn't able to cdef with `tzinfo` (used `object` instead). Would it `tzinfo` have to be `cimport`ed?

**Comment by mroeschke** [comment]
Python 2 builds are failing as expected.

By virtue of the tz_fixtures, `datetime.timezones` are now being tested with

- `DatetimeIndex` & `Timestamp` & `date_range` & `Series` constructor
- `DatetimeIndex` & `Timestamp` methods
- timezone conversion
- indexing
-  comparison and arithmetic ops with arrays & scalars
- `append`/`concat` & other reshaping methods
- `NaT` ops

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/timeseries.rst [hunk 2]
```diff
@@ -2149,12 +2149,9 @@ Time Zone Handling
 ------------------
 
 pandas provides rich support for working with timestamps in different time
-zones using the ``pytz`` and ``dateutil`` libraries.
+zones using the ``pytz`` and ``dateutil`` libraries or ``datetime.timezone``
```
Comment (TomAugspurger) [inline doc/source/user_guide/timeseries.rst]: I think that :class:`datetime.timezone` will work.

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -25,7 +25,7 @@ Other Enhancements
 - ``Series.str`` has gained :meth:`Series.str.casefold` method to removes all case distinctions present in a string (:issue:`25405`)
 - :meth:`DataFrame.set_index` now works for instances of ``abc.Iterator``, provided their output is of the same length as the calling frame (:issue:`22484`, :issue:`24984`)
 - :meth:`DatetimeIndex.union` now supports the ``sort`` argument. The behaviour of the sort parameter matches that of :meth:`Index.union` (:issue:`24994`)
--
+- :meth:`datetime.timezone` objects are now supported as arguments to timezone methods and constructors (:issue:`25065`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.25.0.rst]: meth -> class (not sure if it matters).

File: pandas/_libs/tslibs/timezones.pyx [hunk 4]
```diff
@@ -23,11 +24,12 @@ from pandas._libs.tslibs.util cimport (
     is_string_object, is_integer_object, get_nat)
 
 cdef int64_t NPY_NAT = get_nat()
+cdef object utc_stdlib = timezone.utc
 
 # ----------------------------------------------------------------------
 
 cpdef inline bint is_utc(object tz):
-    return tz is UTC or isinstance(tz, _dateutil_tzutc)
+    return tz is UTC or isinstance(tz, _dateutil_tzutc) or tz is utc_stdlib
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/timezones.pyx]: Potential micro optimization: I suspect that `tz is utc_stdlib` is relatively cheap compared to `isinstance(tz, _dateutil_tzutc)`.  Maybe swap the order of those two for a better average case? Not sure if this matters.

File: doc/source/user_guide/timeseries.rst (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/user_guide/timeseries.rst]: Looks like there are a few more mentions of what's supported

- Reword L272 to include datetime.timezone
- Add a daetetime.timezone example in the example ~L2181

**Review by TomAugspurger — APPROVED** [review]
LGTM, minor nitpicks if you have a chance.

**Review by TomAugspurger — APPROVED** [review]
LGTM. I'd say merge later today if there are no objections.

File: pandas/_libs/tslibs/timezones.pyx [hunk 5]
```diff
@@ -23,11 +24,12 @@ from pandas._libs.tslibs.util cimport (
     is_string_object, is_integer_object, get_nat)
 
 cdef int64_t NPY_NAT = get_nat()
+cdef object utc_stdlib = timezone.utc
 
 # ----------------------------------------------------------------------
 
 cpdef inline bint is_utc(object tz):
-    return tz is UTC or isinstance(tz, _dateutil_tzutc)
+    return tz is UTC or or tz is utc_stdlib or isinstance(tz, _dateutil_tzutc)
```
Comment (TomAugspurger) [inline pandas/_libs/tslibs/timezones.pyx]: Whoops, an extra `or` here.

**Comment by mroeschke** [comment]
@TomAugspurger This branch will always fail the PY2 builds though (feature introduced in 3.3). I was thinking we wait until those builds are removed (hopefully soonish/after 0.24.2)

File: doc/source/user_guide/timeseries.rst [hunk 6]
```diff
@@ -2197,6 +2194,11 @@ To return ``dateutil`` time zone objects, append ``dateutil/`` before the string
                            tz=dateutil.tz.tzutc())
    rng_utc.tz
 
+   # datetime.timezone
```
Comment (jreback) [inline doc/source/user_guide/timeseries.rst]: can you add a versionadded tag

**Review by jreback — APPROVED** [review]
minor comment, otherwise lgtm.

**Comment by mroeschke** [comment]
This should be good to merge now that the PY2 builds are gone.

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**